tests/test_cleaneval.py

```diff
@@ -0,0 +1,486 @@
+#!/usr/bin/env python2.5
+# encoding: utf-8
+"""
+cleaneval.py v1.0
+
+Simple and fast evaluation of CleanEval-1 tasks (precision, recall, F-score).
+"""
+
+import time
+import sys
+import os
+import string
+import chardet
+import codecs
+import re
+import requests
+sys.path.append('')
+import metahtml
+import trafilatura
+from newspaper import parsers
+from bs4 import BeautifulSoup
+from glob import glob
+from math import *
+from difflib import SequenceMatcher
+from getopt import getopt, GetoptError
+from multiset import Multiset
+
+help_message = '''
+This is cleaneval.py version 1.0 -- Copyright (C) 2008 by Stefan Evert
+
+Usage:  cleaneval.py [options] <texts_dir> <gold_dir> [<align_dir>]
+
+Options:
+  -t    print total precision/recall for all files (micro-averaged)
+        (does not print results for individual files)
+  -n    omit table header (e.g. to combine multiple tables)
+  -s    calculate summary statistics (mean / s.d.) and print on STDERR
+  -a    remove non-ASCII characters before comparison
+  -u    calculate unlabelled segmentation accuracy
+
+Evaluates automatically cleaned files in directory <texts_dir> against
+gold standard files in directory <gold_dir>.  Correspoding files in the 
+two directories must have identical names and there must be no other files
+in these directories.
+
+The script prints a TAB-delimited evaluation table on standard output, which
+can be redirected to a file and read into R, Excel or a similar application.
+Precision, recall and F-score are calculated as percentages of whitespace-
+delimited words.  Accuracy of segment identification is measured by precision,
+recall and F-score for labelled or unlabelled segment marker tags (if the
+option -u is specified, no distinction is made between types <p>, <h> and <l>).
+
+If the third argument is given, full alignments will be written to separate
+files in directory <align_dir>.
+'''
+
+re_find_URL = re.compile('https?:\\/\\/(www\\.)?[-a-zA-Z0-9@:%._\\+~#=]{1,256}\\.[a-zA-Z0-9()]{1,6}\\b([-a-zA-Z0-9()!@:%_\\+.~#?&\\/\\/=]*)') # regex detecting url
+
+def __get_url(raw_html):
+    '''
+    Extract url from a raw html
+    Argument:
+        raw_html    String  Raw downloaded html input
+    Return:
+        url         String  Url found in raw_html. If no url is found, return empty string
+    >>> __get_url('')
+    ''
```

Comment:
This function should return `None` rather than an empty string when there is no url